run-tests.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cd "$(dirname "$0")"
+
+port=9000
+
+print_usage() {
+  cat <<EOF
+usage: $0 [IMAGE...]
+
+examples:
+       $0 "sonar-scanner-cli:4.1"
+EOF
+}
+
+info() {
+  echo "[info] $@"
+}
+
+warn() {
+  echo "[warn] $@" >&2
+}
+
+fatal() {
+  echo "[error] $@" >&2
+  exit 1
+}
+
+require() {
+  local prog missing=()
+  for prog; do
+    if ! type "$prog" &>/dev/null; then
+      missing+=("$prog")
+    fi
+  done
+
+  [[ ${#missing[@]} == 0 ]] || fatal "could not find reqired programs on the path: ${missing[@]}"
+}
+
+wait_for_sonarqube() {
+  local i web_up=no sonarqube_up=no
+
+  for ((i = 0; i < 10; i++)); do
+    info "waiting for web server to start ..."
+    if curl -sI localhost:$port | grep '^HTTP/.* 200'; then
+      web_up=yes
+      break
+    fi
+    sleep 5
+  done
+
+  [[ $web_up == yes ]] || return 1
+
+  for ((i = 0; i < 20; i++)); do
+    info "waiting for sonarqube to be ready ..."
+    if curl -s localhost:$port/api/system/status | grep '"status":"UP"'; then
+      sonarqube_up=yes
+      break
+    fi
+    sleep 10
+  done
+
+  [[ "$sonarqube_up" == yes ]]
+}
+
+test_scanner() {
+  local scanner_finished_successfuly=no container_name=$RANDOM
+
+  echo "sanity check image $1"
+  git clone https://github.com/SonarSource/sonar-scanning-examples.git
+  echo "sonar.projectKey=$1-test" >>"$(pwd)/sonar-scanning-examples/sonarqube-scanner/sonar-project.properties"
+  docker run --network=host --name="$container_name" -it -v "$(pwd)/sonar-scanning-examples/sonarqube-scanner:/usr/src" "$1"
+  docker wait "$container_name"
+  echo "Container $container_name stopped."
+  if docker logs "$container_name" | grep 'INFO: EXECUTION SUCCESS'; then
+    scanner_finished_successfuly=yes
+  fi
+
+  [[ "$scanner_finished_successfuly" == yes ]]
+}
+
+launch_sonarqube() {
+  echo "Starting SonarQube in detached mode..."
+  docker run --network=host -d -p $port:9000 sonarqube
+  wait_for_sonarqube
+  echo "SonarQube has been started."
+}
+
+clean_up() {
+  echo "Cleaning up docker containers..."
+  docker stop $(docker ps -aq)
```

Comment:
hum... this works on an ephemeral machine but is quite agressive when run on a local machine (eg. to locally investigate a failure)

Wouldn't it be better/not-so-much-harder to give the created containers specific names and only stop and delete these?

Reply:
Indeed this could be a problem if someone would want to run it locally. Fixed.